test/integration.js

```diff
@@ -0,0 +1,14 @@
+import Spark from '../lib/spark.js'
+import { test } from 'zinnia:test'
+
+test('integration', async () => {
+  const spark = new Spark()
+  for (let i = 0; i < 10; i++) {
```

Comment:
If you check 3 lines below, there's a `break` that will exit this loop once a retrieval succeeded. We need to retry this, because many retrievals don't succeed.

Reply:
Yeah, I noticed the `break`. Let's continue the discussion in https://github.com/filecoin-station/spark/pull/2#discussion_r1209972141